### PR TITLE
CI for fediblockhole

### PR DESCRIPTION
**Pull request by offbyone**
- build and validate the sdist using hynek/build-and-inspect-python-package
- run pytest tests in each supported Python version
- collect and report coverage
- ensure that a developer install works

### CODE REVIEW

**Comment by offbyone** [comment]
This has been idle for a bit; is there anything you'd like me to do before you merge it, or should I abandon it?

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ version = "0.4.5"
 description = "Federated blocklist management for Mastodon"
 readme = "README.md"
 license = {file = "LICENSE"}
-requires-python = ">=3.6"
+requires-python = ">=3.8"
```
Comment (jpwarren) [inline pyproject.toml:7]: Dropping Python 3.6 support (and therefore Ubuntu 18.04LTS) is really my only concern. Do we have to? 

I'm okay if we do, but we will need to make it clear to admins that this is happening so they don't break their setups with an upgrade if later versions of Python aren't available to them.

File: pyproject.toml [hunk 2]
```diff
@@ -42,4 +40,13 @@ build-backend = "hatchling.build"
 [tool.pytest.ini_options]
 addopts = [
     "--import-mode=importlib",
-]
\ No newline at end of file
+]
+norecursedirs = [
+    "tests/helpers",
```
Comment (jpwarren) [inline pyproject.toml:45]: Good change to move this out of setup.cfg

File: pyproject.toml [hunk 3]
```diff
@@ -42,4 +40,13 @@ build-backend = "hatchling.build"
...
+]
+
+[tool.uv]
```
Comment (jpwarren) [inline pyproject.toml:48]: I think I'm good with adding `uv` as a tool dependency as part of implementing CI. It makes sense here.

**Review by jpwarren — CHANGES_REQUESTED** [review]
Good PR overall, and agreed that adding CI will help automate stuff.

Main concern is dropping support for Python 3.6. It's not a hard blocker, just want to understand how necessary it is. We'll need to clearly communicate to admins that it's getting dropped, and do that in a specific release. I suggest the 0.5.0 release. If we can keep 3.6 support, that'd be preferable, but we will need to drop old versions eventually.

I appreciate you taking the time to set this up.

File: pyproject.toml (hunk 1 above)
Comment (offbyone) [reply pyproject.toml:7]: The code today does not support Python 3.6 or 3.7: 

```
ImportError while importing test module '/Users/offby1/projects/wandering.shop/fediblockhole/tests/test_parser_rapidblockjson.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_parser_rapidblockjson.py:3: in <module>
    from fediblockhole.blocklists import parse_blocklist
src/fediblockhole/__init__.py:14: in <module>
    from .blocklists import Blocklist, BlockAuditList, parse_blocklist
src/fediblockhole/blocklists.py:9: in <module>
    from .const import DomainBlock, BlockSeverity, BlockAudit
src/fediblockhole/const.py:5: in <module>
    from typing import NamedTuple, Optional, TypedDict
E   ImportError: cannot import name 'TypedDict' from 'typing' (/Users/offby1/.local/share/mise/installs/python/3.7.17/lib/python3.7/typing.py)
```

It uses features that were only added in Python 3.8

**Review by jpwarren — APPROVED** [review]
Looks good!

The lock file will probably need an update immediately because `v0.4.6` was just released but we will bump the rev to `v0.5.0` shortly anyhow.

**Pull request merged**